EIPS/eip-7377.md

```diff
@@ -0,0 +1,127 @@
+
+## Motivation
+
+For as long as Ethereum has existed, so have complaints about the user experience of the platform.
+
+Ethereum was designed to be an abstract platform, limited only by the mind of the developers who deploy to it. Smart contract wallets were expected to placate unhappy users. Unfortunately, many years into this journey, the tension is still here.
+
+The ecosystem looks different than it did when when the first smart contract wallets were written. Several application-level standards have mitigated some of the issues. There are large standardization efforts around smart contract wallets, and a big push to make them the default on layer two chains. Everything seems to be moving in the right direction.
+
+One significant problems that hasn't been resolved though is what to do about the EOA users of *today*. Manually migrating assets from an EOA is not a tenable path to converting Ethereum's user base to smart contract wallets. We must provide a mechanism, embedded in the protocol, to migrate. This EIP proposes such mechanism.
+
+## Specification
+
+At the fork block `X`, introduce the MigrationTransaction transaction type.
+
+### MigrationTransaction
+
+#### Definition
+
+| field                  | type      |
+|------------------------|-----------|
+| `chainId`              | `int256`  |
+| `nonce`                | `uint64`  |
+| `maxFeePerGas`         | `int256`  |
+| `maxPriorityFeePerGas` | `int256`  |
+| `gasLimit`             | `uint64`  |
+| `codePtr`              | `address` |
+| `storage`              | `List[Tuple[uint256, uint256]]` |
+| `data`                 | `bytes`   |
+| `value`                | `int256`  |
+| `accessList`           | `List[Tuple[address, List[uint256]]]` |
+| `yParity`              | `uint8`   |
+| `v`                    | `uint256` |
+| `r`                    | `uint256` |
+
+The EIP-2718 `TransactionType` is `0x04` and the `TransactionPayload` is `rlp([chainId, nonce, maxFeePerGas, maxPriorityFeePerGas, gasLimit, codePtr, storageTuples, data, value, accessList, yParity, r, s])`.
+
+The transaction's signature hash is `keccak256(0x04 || rlp([chainId, nonce, maxFeePerGas, maxPriorityFeePerGas, gasLimit, codePtr, storageTuples, data, value, accessList])`
+
+#### Validation
+
+A MigrationTransaction is considered valid if the follow properties hold:
+
+* all [EIP-1559](./eip-1559.md) properties, unless specified otherwise
+* the code at `codePtr` is less than the [EIP-170](./eip-170) limit of `24576`
+* the code at `codePtr` must not have size `0`
+
+The intrinsic gas calculation modified from [EIP-1559](./eip-1559) to be `21000 + 16 * non-zero calldata bytes + 4 * zero calldata bytes + 1900 * access list storage key count + 2400 * access list address count + 15000 * length of storage`.
+
+#### Processing
+
+Execution a MigrationTransaction transaction has two parts.
```

Comment:
```suggestion
Executing a `MigrationTransaction` transaction has two steps.
```
```suggestion
Execution of a `MigrationTransaction` transaction has two parts.
```
This is currently grammatically incorrect, and there is a missing code tag. I don't know which fixed version you would prefer, so here they both are.